crates/crypto/bls/src/attestation.rs

```diff
@@ -185,3 +185,78 @@ impl G1Point {
         )
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ark_ff::UniformRand;
+    use rand::{thread_rng, RngCore};
+
+    #[tokio::test]
+    async fn test_keypair_generation() {
+        let mut rng = thread_rng();
+        let private_key = Fr::rand(&mut rng);
+        let keypair = KeyPair::new(private_key).unwrap();
+        let pub_key = keypair.get_pub_key_g1();
+
+        // Check that the public key is not zero
+        assert_ne!(pub_key, G1Projective::zero());
```

Comment:
curious to know how asserting the pub key as not zero helps us. Is this related to any arkworks library thing?